tools/Dockerfile

```diff
@@ -1,93 +1,94 @@
-        \
-    ########### DOTNet Deps ########### \
-        && apt-get update && apt-get install -y --no-install-recommends \
-            software-properties-common \
-            libc6 \
-            libcurl3 \
-            libgcc1 \
-            libgssapi-krb5-2 \
-            libicu52 \
-            liblttng-ust0 \
-            libssl1.0.0 \
-            libstdc++6 \
-            libunwind8 \
-            libuuid1 \
-            zlib1g \
-            \
-    ########### Gateway Deps ########### \
-            libglib2.0-dev \
-            ca-certificates \
-            maven \
-            python2.7 \
-            build-essential \
-            curl \
-            libcurl4-openssl-dev \
-            git \
-            cmake \
-            libssl-dev \
-            uuid-dev \
-            valgrind \
-            wget \
-            \
-    ########### Cleanup ########### \
-        && echo "===> clean up..." \
-        && rm -rf /var/cache/oracle-jdk8-installer \
-        && apt-get clean \
-        && rm -rf /var/lib/apt/lists/*
+#### Update package sources
+    ## For Java
+RUN echo "deb http://ppa.launchpad.net/webupd8team/java/ubuntu trusty main" | tee /etc/apt/sources.list.d/webupd8team-java.list \
+    && echo "deb-src http://ppa.launchpad.net/webupd8team/java/ubuntu trusty main" | tee -a /etc/apt/sources.list.d/webupd8team-java.list \
+    && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys EEA14886
+    ## For CMake
+RUN echo "deb http://ftp.debian.org/debian jessie-backports main" | tee /etc/apt/sources.list.d/cmake.list
+RUN apt-get update
 
-    ENV JAVA_HOME /usr/lib/jvm/java-8-oracle
+#### Install Java
+RUN mkdir -p /usr/share/man/man1/ \
+    && echo debconf shared/accepted-oracle-license-v1-1 select true | debconf-set-selections \
+    && echo debconf shared/accepted-oracle-license-v1-1 seen true | debconf-set-selections \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y oracle-java8-installer oracle-java8-set-default
+ENV JAVA_HOME /usr/lib/jvm/java-8-oracle
 
-########### Install .NET Core SDK ###########
-    ENV DOTNET_SDK_VERSION 1.0.1
-    ENV DOTNET_SDK_DOWNLOAD_URL https://dotnetcli.blob.core.windows.net/dotnet/Sdk/$DOTNET_SDK_VERSION/dotnet-dev-debian-x64.$DOTNET_SDK_VERSION.tar.gz
+#### Install .NET Core dependencies
+RUN apt-get install -y --no-install-recommends \
+        software-properties-common \
+        libc6 \
+        libcurl3 \
+        libgcc1 \
+        libgssapi-krb5-2 \
+        libicu52 \
+        liblttng-ust0 \
+        libssl1.0.0 \
+        libstdc++6 \
+        libunwind8 \
+        libuuid1 \
+        zlib1g
 
-    RUN curl -SL $DOTNET_SDK_DOWNLOAD_URL --output dotnet.tar.gz \
-        && mkdir -p /usr/share/dotnet \
-        && tar -zxf dotnet.tar.gz -C /usr/share/dotnet \
-        && rm dotnet.tar.gz \
-        && ln -s /usr/share/dotnet/dotnet /usr/bin/dotnet
+#### Install Libuv dependencies
+RUN apt-get install -y automake libtool \
```

Comment:
RUN creates a "commit" in the docker image. Do these two packages need to be in a separate commit, or is this just an unintended consequence of the comment?

A snippet from the [documentation](https://docs.docker.com/engine/reference/builder/#run):

> The RUN instruction will execute any commands in a new layer on top of the current image and commit the results. The resulting committed image will be used for the next step in the Dockerfile.
Layering RUN instructions and generating commits conforms to the core concepts of Docker where commits are cheap and containers can be created from any point in an image’s history, much like source control.